### PR TITLE
add option to expose hystrix metrics to statd collector

### PR DESCRIPTION
**Pull request by darshanime**
add `WithStatsDCollector` option to expose hystrix metrics to statsD collector.

sample metrics:
```
myapp.hystrix.MyCommand.totalDuration,metric_type=timing,host=my_host mean=1.1713799999999999,stddev=0.009753362736342112,upper=1.184923,lower=1.162343,count=3i,90_percentile=1.184923,95_percentile=1.184923,99_percentile=1.184923 1591102520000000000
myapp.hystrix.MyCommand.runDuration,metric_type=timing,host=my_host 99_percentile=0,mean=0,stddev=0,upper=0,lower=0,count=3i,90_percentile=0,95_percentile=0 1591102520000000000
myapp.hystrix.MyCommand.errors,metric_type=counter,host=my_host value=1i 1591102520000000000
myapp.hystrix.MyCommand.failures,metric_type=counter,host=my_host value=0i 1591102520000000000
myapp.hystrix.MyCommand.shortCircuits,metric_type=counter,host=my_host value=0i 1591102520000000000
myapp.hystrix.MyCommand.successes,host=my_host,metric_type=counter value=0i 1591102520000000000
myapp.hystrix.MyCommand.timeouts,metric_type=counter,host=my_host value=1i 1591102520000000000
myapp.hystrix.MyCommand.fallbackFailures,metric_type=counter,host=my_host value=0i 1591102520000000000
```

### CODE REVIEW

File: examples/client.go [hunk 1]
```diff
@@ -53,6 +53,7 @@ func hystrixClientUsage() error {
 		hystrix.WithErrorPercentThreshold(25),
 		hystrix.WithSleepWindow(10),
 		hystrix.WithRequestVolumeThreshold(10),
+		hystrix.WithStatsDCollector("localhost:8125", "myapp.hystrix"),
```
Comment (rShetty) [inline examples/client.go:56]: Can we update README too with the details around using this plugin?

File: README.md [hunk 2]
```diff
@@ -103,13 +103,16 @@ client := hystrix.NewClient(
...
 	hystrix.WithErrorPercentThreshold(20),
+	hystrix.WithStatsDCollector("localhost:8125", "myapp.hystrix"),
 )
 
 // The rest is the same as the previous example
 ```
 
 In the above example, there are two timeout values used: one for the hystrix configuration, and one for the HTTP client configuration. The former determines the time at which hystrix should register an error, while the latter determines when the client itself should return a timeout error. Unless you have any special requirements, both of these would have the same values.
 
+You can choose to export hystrix metrics to a statsD collector with the `hystrix.WithStatsDCollector(<statsd addr>, <metrics-prefix>)` option when initializing the client as shown above.
```
Comment (rShetty) [inline README.md:114]: Can we add this in `Plugins` section? Check the current documentation.

Rest all LGTM

Comment (darshanime) [reply README.md:114]: This is not implemented as a plugin but as core functionality that can be activated via option methods. Do you recommend this to be moved to plugins nonetheless?

Comment (rShetty) [reply README.md:114]: Was thinking it would be cleaner to implement it as a plugin. Wdyt?

Comment (darshanime) [reply README.md:114]: hmm, currently we are just using the hystrix-go plugin for this `plugins.InitializeStatsdCollector(client.statsD)`.   
Implementing it as a heimdall plugin would mean registering an in-memory collector with hystrix-go get the data, then triggering the statsd publish on request `OnRequestEnd`.  
Not ideal for 2 reasons:
- we will have to write the in-memory plugin code in heimdall
- the metrics data publishing is not tied to the request flow, but time - so the heimdall plugin interface isn't a good fit for this imo
```go
type Plugin interface {
	OnRequestStart(*http.Request)
	OnRequestEnd(*http.Request, *http.Response)
	OnError(*http.Request, error)
}
```

**Pull request merged**